frontend/src/api/mockAPI/assignments.js

```diff
@@ -316,12 +317,21 @@ class ActiveOffer extends MockAPIController {
                 )}`
             );
         }
-        return this.find(
+
+        const newOffer = this.find(
             this.create({
                 assignment_id: matchingAssignment.id,
-                status: "pending"
+                status: "provisional"
             })
         );
+
+        if (!this.data.offer) {
+            this.data.offers = [newOffer];
```

Comment:
Make sure `.offers` is in the original data-structure and then we can remove this.

Reply:
sure, I 'll have to modify the constructor then?

Reply:
No...it should just be there. In `data.js`

Reply:
I thought it was but if I remove the check it throws an error and if I console log it, it says undefined. I ll take a closer look later.

Reply:
ok, I found a more proper way to update this.